usagov/usagov-2021
# USAGOV-869: Add acronyms to federal agency records

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-869

## Description

 
Acronym content/display functionality: 
* Adds an acronym field to the (Federal) Directory Record content type
* Adds that field to the view for the Federal Agencies directory 
* Adds the field_acronym to various twig files where it should show up
* Adds a preprocess hook to add the acronym, in parentheses, as a suffix to the Title so it can appear in the H1 of an agency full page. _I wonder whether I'm missing a simpler way to do this!_

Acronym import: 
* Adds a form for importing acronyms from a CSV file. This will be a one-time process. 

Cleanup: 
* Removed the form for deleting all Directory Records. That was for use during development. 
* Removed a block that shows up as "broken" from the block layout (block.views_block__wizard_step_test_block_1 -- surely this was a temporary thing)

### TO TEST

Be sure to import configuration and reload caches! 

This assumes your local dev database has Federal agency data, which it probably does if you've refreshed from production within the past few months.

In the CMS, Federal Directory Records now have an "Acronym" text field.  

Download the "acronyms.csv" file attached to the Jira ticket. Log in to the CMS as an admin user and navigate to Admin -> Configuration -> Import acronyms. Select the acronyms.csv file and submit the form. 

Spot-check the A-Z index at http://localhost/agency-index . Some agencies should have an acronym in parentheses after the title. Expand the accordion for an agency with an acronym and click on "More information about ..." You should see the acronym in parentheses in the title of this page as well. 

Also check an agency that does not have an acronym; its display should look normal. 

## Checklist for the Developer



## Checklist for the Peer Reviewers
